Tests/ReactiveSwiftTests/SignalProducerSpec.swift

```diff
@@ -3012,7 +3012,7 @@ class SignalProducerSpec: QuickSpec {
 					.combineLatest(SignalProducer<Int, NoError>.never.promoteError(),
 					               SignalProducer<Double, TestError>.never,
 					               SignalProducer<Float, NoError>.never.promoteError(),
-					               SignalProducer<UInt, POSIXError>.never.flatMapError { _ in .empty })
+					               SignalProducer<UInt, POSIXError>.never.flatMapError { _ in SignalProducer.empty })
```

Comment:
I'd say this is a regression. I use this very often, so it's unfortunate it doesn't compile anymore... I wonder if adding `static var empty` to the protocol would still allow this?

Reply:
Do you mean adding `static var empty` to the `SignalProducerConvertible`?
I think it can't be resolved without a contextual types😢

Reply:
@NachoSoto 
One of solution is to overload `SignalProducer.flatMapError` as below.
```swift
public func flatMapError<Inner: SignalProducerConvertible>(_ transform: @escaping (Error) -> Inner) -> SignalProducer<Value, Inner.Error> where Inner.Value == Value {
    return flatMapError { transform($0).producer }
}

public func flatMapError<NewError>(_ transform: @escaping (Error) -> SignalProducer<Value, NewError>) -> SignalProducer<Value, NewError> {
    return SignalProducer<Value, NewError> { observer, lifetime in
        let serialDisposable = SerialDisposable()
        lifetime += serialDisposable

        self.startWithSignal { signal, signalDisposable in
	    serialDisposable.inner = signalDisposable

	    _ = signal.observeFlatMapError(transform, observer, serialDisposable)
        }
    }
}
```

Reply:
@ra1028 I think it is a reasonable solution & should be applied to existing operators anticipating `SignalProducerConvertible` too. Contextual lookup plays an important role in expressivity.

Reply:
@andersio 
Such protocol works to occur errors like the below.
```swift
extension SignalProducerConvertible {
    /// A producer for a Signal that immediately completes without sending any values.
    public static var empty: SignalProducer<Value, Error> {
        return .empty
    }
}
```
<img width="700" alt="screen shot 2018-02-28 at 19 10 13" src="https://user-images.githubusercontent.com/7347118/36782186-7d52f136-1cbb-11e8-9d33-3fec24adb05f.png">

Therefore, I'll overload existing operators anticipating `SignalProducerConvertible`.

Reply:
I added an overloaded operator to `flatMap` and` flatMapError` for contextual lookup.
Is it also necessary for other operators that will passed `SignalProducerConvertible`?
IMO, there aren't many cases to pass `empty` directly like below.
```swift
producer.then(.empty)
```

Reply:
It is not just `.empty` but also `.init` which is handful when contextual lookup works.

Reply:
@andersio 
done 👍